src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -194,11 +194,13 @@ void Lowering::LowerStoreLoc(GenTreeLclVarCommon* storeLoc)
                 }
             }
 
-            // A local stack slot is at least 4 bytes in size, regardless of
-            // what the local var is typed as, so auto-promote it here
-            // unless it is a field of a promoted struct
-            // TODO-CQ: if the field is promoted shouldn't we also be able to do this?
-            if (!varDsc->lvIsStructField)
+            // Most locals have their own stack slots of a least 4 bytes in
+            // size. Use a wider store in that case. The exceptions are
+            // (dependently) promoted struct fields and small parameters passed
+            // on stack on macOS arm64 ABI.
+            // TODO-CQ: if the field is promoted independently shouldn't we
+            // also be able to do this?
+            if (!varDsc->lvIsStructField && (!varDsc->lvIsParam || varDsc->lvIsRegArg || !compMacOsArm64Abi()))
```

Comment:
> I would probably rather introduce `LclVarDesc::lvStackSlotSize()` and use it from here and from `GetActualRegisterType()`.

I fully agree.

(`GetActualRegisterType()` has become not "actual", and it is unsatisfying)

Reply:
@sandreenko Do you think it would make sense to rename `GetActualRegisterType` to `GetStackSlotType`? What are the semantics behind this function?

Reply:
It does look like this function is used mainly when moving register locals between register and stack, so I think `GetStackSlotType` or maybe `GetCanonicalStackSlotType` (since it will never return e.g. `TYP_FLOAT`) would be a good name.

Reply:
I ended up with `GetStackSlotHomeType`.

Reply:
Thanks for the renaming, it makes the intent clearer.